pandas/core/arrays/datetimelike.py

```diff
@@ -602,10 +602,9 @@ def _validate_listlike(self, value, allow_object: bool = False):
             pass
 
         elif not type(self)._is_recognized_dtype(value.dtype):
-            raise TypeError(
-                f"value should be a '{self._scalar_type.__name__}', 'NaT', "
-                f"or array of those. Got '{type(value).__name__}' instead."
-            )
+            msg = self._validation_error_message(value, True)
```

Comment:
Nice de-duplication.
I suggest using the named argument ``allow_listlike=True`` for clarity.
Otherwise, looks good to me.

Reply:
@jbrockmendel yeah might be more clear (but ok to do in next cleanup PR)